assets/js/vue-apps/form-components.js

```diff
@@ -17,8 +19,117 @@ function initCharacterCount() {
     }
 }
 
+function initBudgetInput() {
+    const mountEl = document.getElementById('js-vue-budget');
+    if (!mountEl) {
+        return;
+    }
+
+    new Vue({
+        el: mountEl,
+        delimiters: ['<%', '%>'],
+        data() {
+            return {
+                maxItems: null,
+                maxBudget: null,
+                items: [],
+                fieldName: null,
+                ready: false,
+                error: null
+            };
+        },
+        watch: {
+            items: {
+                handler() {
+                    const lastItem = this.items[this.items.length - 1];
+                    if (lastItem) {
+                        if (this.items.length < this.maxItems && (!isEmpty(lastItem.item) || !isEmpty(lastItem.cost))) {
+                            this.addItem();
+                        }
+                    }
+                    if (this.items.length === this.maxItems) {
+                        this.setError('TOO_MANY_ITEMS');
+                    } else {
+                        this.clearError();
+                    }
+                },
+                deep: true
+            }
+        },
+        mounted: function() {
+            // Clear out <noscript> fallbacks otherwise they submit as well as the
+            // Vue-enhanced fields (eg. double inputs)
+            $(this.$el)
```

Comment:
Interesting. It might be worth comparing this to how the past grants works. There's some Vue behaviour where content inside the component or vue scope will get replaced with any new content.

Reply:
That wasn't explained well, will see if I can find an example of what I mean.

Reply:
Got this to work – the trick is to use a self-closing element, which seems to then consume anything following the component (but not before). Hope this isn't a bug...